llvm/lib/Target/LoongArch/LoongArchTargetTransformInfo.cpp

```diff
@@ -19,6 +19,20 @@ using namespace llvm;
 
 #define DEBUG_TYPE "loongarchtti"
 
+bool LoongArchTTIImpl::areInlineCompatible(const Function *Caller,
+                                           const Function *Callee) const {
+  const TargetMachine &TM = getTLI()->getTargetMachine();
+
+  const FeatureBitset &CallerBits =
+      TM.getSubtargetImpl(*Caller)->getFeatureBits();
+  const FeatureBitset &CalleeBits =
+      TM.getSubtargetImpl(*Callee)->getFeatureBits();
+
+  // Inline a callee if its target-features are a subset of the callers
+  // target-features.
+  return (CallerBits & CalleeBits) == CalleeBits;
+}
```

Comment:
Can we make this the default implementation? I think every target copies some variant of this...

Reply:
LG for LoongArch.
Agree to try making it the default implementation.

Reply:
New PR for this: https://github.com/llvm/llvm-project/pull/117493